buildSrc/src/main/groovy/airbyte-bulk-connector.gradle

```diff
@@ -162,15 +173,51 @@ class AirbyteBulkConnectorPlugin implements Plugin<Project> {
             dependsOn project.tasks.matching { it.name ==~ /(compile|spotbugs)[a-zA-Z]*Java/ }
         }
 
-        project.tasks.register('integrationTestJava', Test) {
-            description = 'Runs the integration tests.'
+        // This task exists solely so that we can run the "modern" integration tests
+        // in non-docker mode on our laptops.
+        // It's not a dependency of any other task.
+        project.tasks.register('integrationTestNonDocker', Test) {
+            description = 'Runs the integration tests in non-docker mode.'
+            group = 'verification'
+            testClassesDirs = project.sourceSets.integrationTest.output.classesDirs
+            classpath = project.sourceSets.integrationTest.runtimeClasspath
+            useJUnitPlatform()
+            mustRunAfter project.tasks.check
+        }
+        // This task runs the "modern" integration tests in docker mode.
+        // It's intended for use in CI.
+        project.tasks.register('integrationTestDocker', Test) {
+            description = 'Runs the integration tests in docker mode.'
             group = 'verification'
-            testClassesDirs = project.sourceSets.integrationTestJava.output.classesDirs
-            classpath = project.sourceSets.integrationTestJava.runtimeClasspath
+            testClassesDirs = project.sourceSets.integrationTest.output.classesDirs
+            classpath = project.sourceSets.integrationTest.runtimeClasspath
             useJUnitPlatform()
             mustRunAfter project.tasks.check
+            // We need a docker image to run this task, so depend on assemble
+            dependsOn project.tasks.assemble
+            environment "MICRONAUT_ENVIRONMENTS", "DOCKERIZED_INTEGRATION_TEST"
```

Comment:
(see https://docs.micronaut.io/latest/guide/#environments - arguably this should be `systemProperty "micronaut.environments"` instead of `environment "MICRONAUT_ENVIRONMENTS"` 🤔 )

Reply:
Don't feel like you _should_ but you could consider using the `FeatureFlag` functionality I'm about to add to set this `DOCKERIZED_INTEGRATION_TEST` environment. The advantage in doing so is that you don't interfere with `MICRONAUT_ENVIRONMENTS` leaving it free to be used in a different way.

Reply:
I think that's orthogonal - your pr will allow test writers to launch the connector with different envs; this is so that I can run the same test code (`src/test-integration`) under different envs

(see the NonDockerizedDestination.kt diff for what exactly we're doing)